client/components/docs/docs.service.js

```diff
@@ -1,48 +1,51 @@
-class DocsService {
-    constructor ($translate, TranslateService, TARGET, DOCS_ALL_GUIDES, DOCS_HOMEPAGE_GUIDES) {
-        this.$translate = $translate;
-        this.TranslateService = TranslateService;
-        this.TARGET = TARGET;
-        this.DOCS_ALL_GUIDES = DOCS_ALL_GUIDES;
-        this.DOCS_HOMEPAGE_GUIDES = DOCS_HOMEPAGE_GUIDES;
-    }
+{
+    class DocsService {
+        constructor ($translate, OvhApiMe, TranslateService, TARGET, DOCS_ALL_GUIDES, DOCS_HOMEPAGE_GUIDES) {
+            this.$translate = $translate;
+
+            this.OvhApiMe = OvhApiMe;
+            this.TranslateService = TranslateService;
 
-    getDomainOfGuides () {
-        if (this.TARGET === "US") {
-            return "US";
+            this.TARGET = TARGET;
+            this.DOCS_ALL_GUIDES = DOCS_ALL_GUIDES;
+            this.DOCS_HOMEPAGE_GUIDES = DOCS_HOMEPAGE_GUIDES;
         }
-        const locale = this.TranslateService.getGeneralLanguage();
 
-        if (locale === "fr") {
-            return "FR";
+        fetchSubsidiary () {
+            return this.OvhApiMe.v6()
+                .get().$promise
+                .then(me => me.ovhSubsidiary);
         }
-        return "EN";
-    }
 
-    getAllGuidesLink () {
-        const domain = this.getDomainOfGuides();
-        switch (domain) {
-            case "US":
-                return this.DOCS_ALL_GUIDES.US;
-            case "FR":
-                return this.DOCS_ALL_GUIDES.FR;
-            case "EN":
-            default:
-                return this.DOCS_ALL_GUIDES.EN;
+        fetchRootGuidePageLink () {
+            return this.fetchSubsidiary()
+                .then(subsidiaryName => {
+                    const matchingGuide = this.DOCS_ALL_GUIDES[subsidiaryName];
+
+                    return _(matchingGuide).isString() ? matchingGuide : this.DOCS_ALL_GUIDES.DEFAULT;
+                });
         }
-    }
 
-    getGuidesOfSection (section) {
-        const domain = this.getDomainOfGuides();
-        const sectionContent = this.DOCS_HOMEPAGE_GUIDES[domain][section];
+        fetchGuidePageLinkFor (sectionName) {
```

Comment:
GuidePageLinkFor -> I have no idea what this function is supposed to do just by looking at its name. My only guess is that it fetches a single guide page for a certain section.

Reply:
It fetchs the link for a guide page for a certain section according to the section name😄